src/features/health/region.controller.js

```diff
@@ -1,17 +1,34 @@
 const { Region } = require('../../shared/db/mongodb/schemas');
+const {getTopAgentsBySales} = require('../../../src/features/health/agent.controller')
 
 const createRegion = async (req, res) => {
-    const region = req.query.region;
+    const regionName = req.query.region;
+    console.log(`Checking for existing region: ${regionName}`);
 
     // checking to see if region exist
-    const existingRegion = await Region.findOne({name: regionName});
+    const existingRegion = await Region.findOne({region: regionName});
+    console.log(`Existing region found: ${existingRegion}`);
+
     if (existingRegion) {
-        return res.status(400).send({ message: `Region ${regionName} already exists`});
+        return res.status(400).send({ message: "Region already exists"});
     }
+
     // creating new region
     const newRegion = new Region({name: regionName});
```

Comment:
Still early on in development of this endpoint, but make sure to actually fill in in the information about the region in some way.

Reply:
Will do!